test/set/pathValues/atom.spec.js

```diff
@@ -167,6 +167,25 @@ describe("an atom", function() {
         }));
     });
 
+    it("through a reference that lands on an atom with a null last key", function() {
+
+        var cache = {};
+        var version = 0;
+        setPathValues(
+            getModel({ cache: cache, version: version++ }), [
+                $pathValue("foo.value", $ref("bar.value")),
+                $pathValue("bar.value", $atom("bar-value")),
+                $pathValue(["foo", "value", null], $atom("foo-value-1")),
```

Comment:
Just getting around to looking at this. Sorry about the delay.

What's the motivation in setting `["foo", "value", null]` twice as part of the test?

Trying to make sure I understand the change. I believe I understand the second set to foo-value-2, and the subsequent assertion (to make sure the reference is maintained, and the atom is the thing which actually gets set/overwritten), but I didn't follow what the first set to foo-value-1 is trying to validate.